### PR TITLE
Add PUBLIC_KEY_URL option

### PR DESCRIPTION
**Pull request by xezpeleta**
The public key will be downloaded from the environment variable PUBLIC_KEY_URL

[linuxserverurl]: https://linuxserver.io
[linuxserver.io][linuxserverurl]

------------------------------

 - [x] I have read the [contributing](https://github.com/linuxserver/docker-openssh-server/blob/master/.github/CONTRIBUTING.md) guideline and understand that I have made the correct modifications

------------------------------

## Description:
The following PR adds the option to download the public key from a URL, using the environment variable "PUBLIC_KEY_URL"

## Benefits of this PR and context:
For instance you could add the following environment variable: `PUBLIC_KEY_URL=https://github.com/username.keys` so you don't need to copy-paste the keys or copy the files.

## How Has This Been Tested?
I've tested with the example docker-compose.yml

### CODE REVIEW

File: root/etc/cont-init.d/50-config [hunk 1]
```diff
@@ -65,6 +65,12 @@ fi
     echo "$PUBLIC_KEY" >> /config/.ssh/authorized_keys && \
     echo "Public key from env variable added"
 
+[[ -n "$PUBLIC_KEY_URL" ]] && \
+    PUBLIC_KEY_DOWNLOADED=$(curl -s $PUBLIC_KEY_URL)
```
Comment (aptalca) [inline root/etc/cont-init.d/50-config]: Missing `&& \` at the end

**Review by aptalca — CHANGES_REQUESTED** [review]
Logic looks good except for the missing `&& \`

Please add the var listing as optional to readme-vars.yml, and add a changelog entry

**Comment by xezpeleta** [comment]
Hi! Is this resolved? Or is there any other change missing?

Thanks

**Comment by tobbenb** [comment]
Read aptalcas comment again and you'll see what you need to add.

**Comment by Roxedus** [comment]
To echo aptalca; Please add the var listing as optional to readme-vars.yml, and add a changelog entry

**Review by aptalca — APPROVED** [review]

**Pull request merged**